bookkeeper-server/src/test/java/org/apache/bookkeeper/client/MockBookKeeperTestCase.java

```diff
@@ -424,11 +494,19 @@ protected void setupBookieClientAddEntry() {
                         ledgerId, entryId, bookieSocketAddress, ctx);
                 } else {
                     if (getMockLedgerContentsInBookie(ledgerId, bookieSocketAddress).isEmpty()) {
```

Comment:
@ArvinDevel 
before this line put
{code}
if (failedBookies.contains(bookieSocketAddress)) {
callback.writeComplete(NoBookieAvailableException, ledgerId, entryId, bookieSocket...)
}
{code}

and drop the check in registerMockEntryForRead

Reply:
@eolivelli thanks, fixed.